amgut/lib/data_access/test/test_survey.py

```diff
@@ -0,0 +1,452 @@
+        c = ascii_letters + '1234567890'
+        notes_test = ''.join([choice(c) for i in range(40)])
+
+        # Set up survey
+        survey_id = '817ff95701f4dd10'
+        survey = Survey(2)
+        consent = {
+            'login_id': 'eba20873-b7db-33cc-e040-8a80115d392c',
+            'survey_id': survey_id,
+            'participant_name': 'some name that should be ignored',
+            'age_range': 'ANIMAL_SURVEY',
+            'parent_1_name': 'ANIMAL_SURVEY',
+            'parent_2_name': 'ANIMAL_SURVEY',
+            'deceased_parent': False,
+            'participant_email': 'REMOVED',
+            'obtainer_name': 'ANIMAL_SURVEY',
+            'assent_obtainer': 'ANIMAL_SURVEY',
+            'is_juvenile': True}
+        with_fk = [(survey_id, 128, 'Other'),
+                   (survey_id, 129, 'Wild'),
+                   (survey_id, 131, 'Male'),
+                   (survey_id, 132, 'Suburban'),
+                   (survey_id, 133, 'Normal'),
+                   (survey_id, 134, 'Omnivore'),
+                   (survey_id, 135, 'Wild food'),
+                   (survey_id, 136, 'Both'),
+                   (survey_id, 137, 'Unspecified'),
+                   (survey_id, 138, 'Lives alone with humans'),
+                   (survey_id, 139, '8+'),
+                   (survey_id, 140, 'Unspecified'),
+                   (survey_id, 141, 'Never')]
+        without_fk = [(survey_id, 130, '["20"]'),
+                      (survey_id, 142, '["Giant ratty pet!"]'),
+                      (survey_id, 143, '["Capybara"]'),
+                      (survey_id, 144, '["%s"]' % notes_test),
+                      (survey_id, 145, '["29 - Male"]'),
+                      (survey_id, 127, '["Fluffy"]')]
+
+        survey.store_survey(consent, with_fk, without_fk)
+        obs = survey.fetch_survey(survey_id)
+        exp = {'Pet_Information_127_0': 'Fluffy',
+               'Pet_Information_137_0': [0],
+               'Pet_Information_143_0': 'Capybara',
+               'Pet_Information_132_0': 2,
+               'Pet_Information_138_0': 1,
+               'Pet_Information_144_0': notes_test,
+               'Pet_Information_129_0': 4,
+               'Pet_Information_142_0': 'Giant ratty pet!',
+               'Pet_Information_133_0': 3,
+               'Pet_Information_139_0': 5,
+               'Pet_Information_130_0': '20',
+               'Pet_Information_128_0': 9,
+               'Pet_Information_134_0': 2,
+               'Pet_Information_140_0': 0,
+               'Pet_Information_131_0': 1,
+               'Pet_Information_145_0': '29 - Male',
+               'Pet_Information_136_0': 3,
+               'Pet_Information_135_0': [3],
+               'Pet_Information_141_0': 4}
+        self.assertEqual(obs, exp)
+
+        obs = ag_data.getConsent(survey_id)
+        consent['ag_login_id'] = consent['login_id']
+        del consent['login_id']
+        del consent['obtainer_name']
```

Comment:
The `del` statements are a little bothersome.  Personally, I'd prefer to create a completely new object, since its already be heavily modified.

But this is just personal preference - definitely not blocking